konnectivity-client/pkg/client/client.go

```diff
@@ -95,7 +95,31 @@ func (t *grpcTunnel) serve(c clientConn) {
 	defer c.Close()
 
 	for {
-		pkt, err := t.stream.Recv()
+		// It is possible for stream.Recv() to block indefinitely, especially when the stream is full.
+		// Workaround is to run stream.Recv() in a separate goroutine with a generous timeout to prevent
+		// goroutine leaks. See https://github.com/grpc/grpc-go/issues/1229 for more details.
+		done := make(chan struct{})
+		pktChan := make(chan *client.Packet, 1)
+		errChan := make(chan error, 1)
+		go func() {
```

Comment:
Do we really want to spin up a separate goroutine for each packet we want to receive?

Reply:
It's not ideal but it's a side effect of stream.Recv not accept a context.

I'm on the fence about keeping this change though, I pushed it so I can test it but I haven't seen any evidence yet that it helps with the issue.